src/subscription/subscribe.js

```diff
@@ -143,16 +151,23 @@ function subscribeImpl(
   // the GraphQL specification. The `execute` function provides the
   // "ExecuteSubscriptionEvent" algorithm, as it is nearly identical to the
   // "ExecuteQuery" algorithm, for which `execute` is also used.
-  const mapSourceToResponse = payload =>
-    execute({
+  // If `perEventContextResolver` is provided, it is invoked with the original
+  // `contextValue` to return a new context unique to this `execute`.
+  const mapSourceToResponse = payload => {
+    const executeContextValue =
+      typeof perEventContextResolver === 'function'
+        ? perEventContextResolver(contextValue)
+        : contextValue;
+    return execute({
```

Comment:
It's better to do a single check per stream:
```js
const perEventContextResolverFn = perEventContextResolver ?? (ctx) => ctx
```

Reply:
Updated to a single check per-stream, kept with the `typeof ` check rather than the `??` just because it's more explicit.